metaflow/datastore/task_datastore.py

```diff
@@ -553,6 +555,25 @@ def done(self):
         )
 
         if self._metadata:
+            origin_task_metadata = []
+            if self._origin_task_id is not None:
```

Comment:
This is probably not the right place for this. Why not place it in the run_step method where we currently set origin-run-id?

Reply:
One reason for placing it there was because [clone_only](https://github.com/Netflix/metaflow/blob/584b7e1996fa1798a1cb9d6ede03ecf145d7ddc4/metaflow/task.py#L238) method was exposed by the `MetaflowTask`. Since in the CLI we call `clone_only` [or](https://github.com/Netflix/metaflow/blob/584b7e1996fa1798a1cb9d6ede03ecf145d7ddc4/metaflow/cli.py#L535) `run_step`, if we place this in run_step then we are not getting origin-task-id as origin-task-id is available on cloning a past task-datastore and is nonexistent for a freshly run task.  

To move it out of the task_datastore I can add this to the `clone_only` method of the `MetaflowTask`. Is that a better option?

Reply:
As a change, I have moved setting `origin-task-id` into `MetaflowTask.clone_only` instead of `TaskDatastore.done`